### PR TITLE
[WIP] Fix EZP-28087: Cannot use link in RichText editor

### PR DESCRIPTION
**Pull request by adamwojs**
> JIRA: https://jira.ez.no/browse/EZP-28087

## Description
 
~POC for cover a wider range of URLs formats used in Richtext editor.~

PR extending the range of supported URLs formats in Richtext editor.  

## TODO
- [ ] Gather feedback
- [ ] Clean up code

### CODE REVIEW

File: eZ/Publish/Core/FieldType/RichText/XSLTProcessorFunctions.php [hunk 1]
```diff
@@ -12,19 +12,50 @@
...
-    const URL_PATTERN = '%^((ezlocation|ezurl|ezcontent):\/\/\d+(#[^\s]+)?)|(#[^\s]+)|(?:(?:https?|ftps?)://)(?:\S+(?::\S*)?@|\d{1,3}(?:\.\d{1,3}){3}|(?:(?:[a-z\d\x{00a1}-\x{ffff}]+-?)*[a-z\d\x{00a1}-\x{ffff}]+)(?:\.(?:[a-z\d\x{00a1}-\x{ffff}]+-?)*[a-z\d\x{00a1}-\x{ffff}]+)*(?:\.[a-z\x{00a1}-\x{ffff}]{2,6}))(?::\d+)?(?:[^\s]*)?$%iu';
+    const URL_SCHEMA_WHITELIST = [
+        'http',
+        'https',
+        'ftp',
+        'ftps',
+        'ezcontent',
+        'ezlocation',
+        'ezurl',
+    ];
```
Comment (glye) [inline eZ/Publish/Core/FieldType/RichText/XSLTProcessorFunctions.php]: We might include `mailto:`, ref suggestion in https://github.com/ezsystems/ezpublish-kernel-ee/pull/161

**Comment by adamwojs** [comment]
Please hold on for a while with code review, I've found some issue with this approach.

**Comment by adamwojs** [comment]
I've decided to go back to regex because as the original idea was wrong in assumptions http://php.net/manual/en/function.parse-url.php: 

> This function is not meant to validate the given URL, it only breaks it up into the above listed parts. Partial URLs are also accepted, parse_url() tries its best to parse them correctly.

I've overlooked this paragraph yesterday, sorry for that.

I've split validation in two cases, which in my opinion will simplify things:
* values that are absolute URLs
* values that are relative URLs (that begin with path or fragment component)

**Comment by andrerom** [comment]
@adamwojs unfortunately we will need to revert #2090 (in #2124) as the break on mac is a blocker, and this issue is a release blocker for 6.7.6-rc1 and 6.12.0-rc1.

Do you have some thoughts on how to accomplish this without the php functions?

From what I can see, brackets [ or ] are not valid as per RFC 3986, so to somehow escape them is most likely a very valid approach here, otherwise the resulting docbook is basically with invalid urls. But maybe we can do that in a cleaner and more tested way then what was proposed in #2044?

**Pull request closed without merging**